scripts/sorse_zenodo_upload.py

```diff
@@ -0,0 +1,212 @@
+            continue
+        # Create empty deposition - Should return 201
+        logging.info("Creating new deposition for %s", path)
+        r = requests.post(api_uri+'/api/deposit/depositions',
+            params=params,
+            json={},
+            headers=headers)
+        
+        if (r.status_code != 201):
+            logging.error("Failed to create empty deposition! Response: %i: %s", r.status_code, r.content)
+            print("Error processing {}, check log file for more information".format(path))
+            continue
+        # Fetch DOI from prereservation
+        doi = r.json()['metadata']['prereserve_doi']['doi']
+        # Fetch bucket url to put files
+        bucket_url = r.json()["links"]["bucket"]
+        # Fetch deposition id for next requests
+        deposition_id = r.json()['id']
+        logging.info("Received info for new deposition: id: %s, doi: %s, bucket_url: %s", deposition_id, doi, bucket_url)
+        # Fetch authors from frontmatter and resolve affiliations links
+        authors = post['authors']
+        affiliations = post['affiliations']
+        creators = []
+        for aut in authors:
+            creator = dict()
+            creator['name'] = aut['name']
+            if not 'affiliation' in aut:
+                logging.error('Author %s in event %s does not have an affiliation! Defaulting to N/A', aut['name'], path)
+                creator['affiliation'] = 'N/A'
+            else:
+                aff_index = aut['affiliation']
+                # find affiliation string
+                for keyval in affiliations:
+                    if aff_index == keyval['index']:
+                        creator['affiliation'] = keyval['name']
+                        break
+            if 'orcid' in aut:
+                creator['orcid'] = aut['orcid'] 
+            
+            # do not add other author fields, because the Zenodo API will complain
+            creators.append(creator)
+        
+        # update .md with DOI
+        post['doi'] = doi
+        filename, file_extension = os.path.splitext(str(path))
+        outputpath = str(path) if overwrite else filename+'-new'+file_extension
+        output_file = open(outputpath, 'wb')
+        frontmatter.dump(post, output_file)
+        output_file.close()
+        # generate PDF
+        temp_file = open(workingpath + '/' + path.name, 'wb')
+        frontmatter.dump(post, temp_file)
+        temp_file.close()
+        # copyfile(outputpath, './generate-pdf/'+ path.name) # this might not finish before the subprocess call
+        subprocess.call(['sh', './generate-pdfs.sh'])
+        # os.remove('./generate-pdf/'+ path.name)
+
+        # The target URL is a combination of the bucket link with the desired filename
+        # seperated by a slash.
+        pdfpath = workingpath + '/' + os.path.basename(filename) + '.pdf' # use the event filename for the pdf

```

Comment:
Should we change this to the actual pdf destination? e.g downloads? I chose workingpath to make sure I do not accidently overwrite existing data with malformed PDFs.

Reply:
yes, we should move it to the downloads folder

Reply:
and overwrite originals without warning or review ? :smile:

Reply:
we use git, so everything is reversible

Reply:
you got me 🤦